src/main/java/com/linkedin/kmf/services/MultiClusterTopicManagementService.java

```diff
@@ -146,198 +158,198 @@ public void run() {
-      Node leader = leaderOption.isEmpty() ?  null : new Node((Integer) leaderOption.get(), "", -1);
-      Node[] replicas = new Node[scalaTuple._2().size()];
-      for (int i = 0; i < replicas.length; i++) {
-        Integer brokerId = (Integer) scalaTuple._2().apply(i);
-        replicas[i] = new Node(brokerId, "", -1);
+    private static List<PartitionInfo> getPartitionInfo(ZkUtils zkUtils, String topic) {
+      scala.collection.mutable.ArrayBuffer<String> topicList = new scala.collection.mutable.ArrayBuffer<>();
+      topicList.$plus$eq(topic);
+      scala.collection.Map<Object, scala.collection.Seq<Object>> partitionAssignments =
+          zkUtils.getPartitionAssignmentForTopics(topicList).apply(topic);
+      List<PartitionInfo> partitionInfoList = new ArrayList<>();
+      scala.collection.Iterator<scala.Tuple2<Object, scala.collection.Seq<Object>>> it = partitionAssignments.iterator();
+      while (it.hasNext()) {
+        scala.Tuple2<Object, scala.collection.Seq<Object>> scalaTuple = it.next();
+        Integer partition = (Integer) scalaTuple._1();
+        scala.Option<Object> leaderOption = zkUtils.getLeaderForPartition(topic, partition);
+        Node leader = leaderOption.isEmpty() ?  null : new Node((Integer) leaderOption.get(), "", -1);
+        Node[] replicas = new Node[scalaTuple._2().size()];
+        for (int i = 0; i < replicas.length; i++) {
+          Integer brokerId = (Integer) scalaTuple._2().apply(i);
+          replicas[i] = new Node(brokerId, "", -1);
+        }
+        partitionInfoList.add(new PartitionInfo(topic, partition, leader, replicas, null));
       }
-      partitionInfoList.add(new PartitionInfo(topic, partition, leader, replicas, null));
-    }
 
-    return partitionInfoList;
-  }
+      return partitionInfoList;
+    }
 
-  protected static int getReplicationFactor(List<PartitionInfo> partitionInfoList) {
-    if (partitionInfoList.isEmpty())
-      throw new RuntimeException("Partition list is empty");
+    static int getReplicationFactor(List<PartitionInfo> partitionInfoList) {
+      if (partitionInfoList.isEmpty())
+        throw new RuntimeException("Partition list is empty");
 
-    int replicationFactor = partitionInfoList.get(0).replicas().length;
-    for (PartitionInfo partitionInfo : partitionInfoList) {
-      if (replicationFactor != partitionInfo.replicas().length) {
-        String topic = partitionInfoList.get(0).topic();
-        throw new RuntimeException("Partitions of the topic " + topic + " have different replication factor");
+      int replicationFactor = partitionInfoList.get(0).replicas().length;
+      for (PartitionInfo partitionInfo : partitionInfoList) {
+        if (replicationFactor != partitionInfo.replicas().length) {
+          String topic = partitionInfoList.get(0).topic();
+          throw new RuntimeException("Partitions of the topic " + topic + " have different replication factor");
```

Comment:
Yes, I think it is OK to exit. It is actually impossible to see different replication factor for partitions of the same topic if the topic is created using our current API and tool. It is not an intermediate state that can recovery by itself later. Thus TopicManagementService needs to stop for someone to look into the problem.